test/runtests.jl

```diff
@@ -183,9 +200,16 @@ end
         @test  any(x isa AllocationSite && x.type == Memory        # uses jl_genericmemory_copy_slice
                    for x in check_allocs(copy, (Vector{Int},)))
 
-        @test  all(x isa AllocationSite && x.type == Memory{UInt8} # uses jl_string_to_genericmemory
+        @test  all(x isa DynamicDispatch || (x isa AllocationSite && x.type == Memory{UInt8}) # uses jl_string_to_genericmemory
                    for x in check_allocs(Base.array_new_memory, (Memory{UInt8}, Int)))
 
+        # Marked broken because the `Expr(:foreigncall, QuoteNode(:jl_alloc_string), ...)` should be resolved
+        # by AllocCheck.jl, but is instead (conservatively) marked as a DynamicDisaptch.
+        #
+        # We get thrown off by the `jl_load_and_lookup` machinery here.
+        @test_broken all(x isa AllocationSite && x.type == Memory{UInt8} # uses jl_string_to_genericmemory
```

Comment:
I relaxed this test and marked the original as broken

We need to expand our resolution here to walk through the `PhiNode` in the LLVM IR and notice the ccall / `jl_load_and_lookup` machinery.